.github/workflows/dependency_checker.yml

```diff
@@ -0,0 +1,30 @@
+name: Dependency Checker
+on:
+  pull_request:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * 1-5'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v5
+      with:
+        python-version: 3.9
+    - name: Install dependencies
+      run: |
+        python -m pip install .[dev]
+        make check-deps OUTPUT_FILEPATH=latest_requirements.txt
```

Comment:
Just a note, this will output the file in the root directory. We could put this in tests or somewhere else

Reply:
I think that it is fine to be in root but I have no preference on where it should go (kept it consistent with the other dependency checker)
cc: @amontanez24 for input

Reply:
Yeah I'm fine with it